Alpaca.Markets/SockClientBase.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using WebSocket4Net;
+using System.Security.Authentication;
+
+namespace Alpaca.Markets
+{
+    /// <summary>
+    /// Provides unified type-safe access for websocket streaming APIs.
+    /// </summary>
+    public abstract partial class SockClientBase : IDisposable
+    {
+        private protected IWebSocket _webSocket;
```

Comment:
I dislike idea with providing protected field only for event subscription reason. See me previous issue about `OnError` event. For `MessageReceived` and `DataReceived` events we can also write generic empty handlers in this class and override these handlers in `SockClient` and `PolygonSockClient` classes. Moreover in this case we can correctly unsubscribe all events in `Dispose` method.

Reply:
Will do this refactoring in separate commit after this PR merge.